docs/docs/guides/smart_contracts/testing_contracts/index.md

```diff
@@ -0,0 +1,184 @@
+---
+title: Testing Contracts
+---
+
+Aztec contracts can be tested in a variety of ways depending on the needs of a particular application and the complexity of the interactions they must support.
+
+To test individual contract functions, you can use the Testing eXecution Environment (TXE) described below. For more complex interactions that require checking that the protocol rules are enforced, you should [write end-to-end tests using TypeScript](../../js_apps/test.md).
+
+## Pure Noir tests
+
+Noir supports the `#[test]` annotation which can be used to write simple logic tests on isolated utility functions. These tests only make assertions on algorithms and cannot interact with protocol-specific constructs such as `storage` or `context`, but are extremely fast and can be useful in certain scenarios.
+
```

Comment:
Can we add a section on when to use TXE vs end-to-end tests.
Could we add an analogy like: macha+ethers:our e2e :: foundry: TXE

This is my biggest confusion reading the doc right now